src/runtime_src/ert/scheduler/scheduler.cpp

```diff
@@ -623,8 +633,10 @@ configure_cu(addr_type cu_addr, addr_type regmap_addr, size_type regmap_size)
 {
   // write register map, starting at base + 0x10
   // 0x4, 0x8, 0xc used for interrupt, which is initialized in setup
-  for (size_type idx = 4; idx < regmap_size; ++idx)
-    write_reg(cu_addr + (idx << 2), read_reg(regmap_addr + (idx << 2)));
+  uint32_t *addr_ptr = (uint32_t *)(uintptr_t)cu_addr;
+  uint32_t *regmap_ptr = (uint32_t *)(uintptr_t)regmap_addr;
+
+  memcpy(addr_ptr+4, regmap_ptr+4, regmap_size);
```

Comment:
@chienwei-lan , will this break hardware emulation? write_reg() is emulated, but memcpy is not. I'm wondering if we should #ifdef ERT_HW_EMU and keep emulation code as is?